media/js/src/CollectionTab.jsx

```diff
@@ -71,18 +71,24 @@ export default class CollectionTab extends React.Component {
             selectedAsset: null
         });
 
-        if (!this.props.assets || !this.props.assets.length) {
-            const me = this;
-            getAssets().then(function(d) {
-                me.props.onUpdateAssets(
-                    d.assets,
-                    d.asset_count,
-                    d.active_tags,
-                    d.active_vocabulary,
-                    d.space_viewer.id
-                );
-            });
-        }
+        // Always reload the assets when leaving asset defail view.
+        // The user might have made changes to the selections, and
+        // this needs to be up to date.
+
+        // Set assets to null, to display "Loading assets".
+        this.props.onUpdateAssets(null);
+
+        // Reload the assets.
+        const me = this;
+        getAssets().then(function(d) {
```

Comment:
After creating a selection and navigating back to the full collection, the collection refreshes to show all items rather than those from "Me". I'm also wondering if a full refresh is needed, or just a refresh for the asset that was being changed?

Reply:
Hmmm, it's true. Reloading all the assets was the simplest way to do this, but now that I've thought it out, I think it is possible to just setState with a new `assets` array, and include the updated state of the asset. This would avoid an extra ajax call.

Reply:
That might be a good approach...getting the assets is still a bit slow. Based on your comments, Is this PR ready to be merged, or should I wait?

Reply:
I will make that change here in this PR - hold on.